rtc_service/rtc_service.py

```diff
@@ -0,0 +1,288 @@
+        else:
+            self.timezone_offset_s = timezone_offset_s
+            logging.info("Timezone offset is initialized with parameter")
+        logging.info(f"Timezone offset is set to {self.timezone_offset_s}s")
+
+        self.get_time_from_local = get_time_from_local
+        if not self.get_time_from_local:
+            logging.info("RTC time is taken from a ntp server")
+            self.ntp_client = ntplib.NTPClient()
+        else:
+            logging.info("RTC time is taken from local time")
+        self.sink_manager = sink_manager
+        logging.info(f"Expected RTC sending period is set to {self.period}s")
+
+        self.running = False
+
+    def publish_time(self):
+        """
+        Publish the rtc time in the network.
+        """
+        if not self.get_time_from_local:
+            try:
+                req = self.ntp_client.request('pool.ntp.org', version=3)
+                timestamp = req.dest_time+req.offset
+            except ntplib.NTPException as err:
+                logging.warning("Couldn't get time from NTP server. (%s)", err)
+                return
+        else:
+            timestamp = time()
+
+        timestamp_ms = int(timestamp*1000)
+        data_payload = RTC_VERSION.to_bytes(2, "little") + encode_tlv(timestamp_ms, self.timezone_offset_s)
+        logging.info("Send rtc message to the network")
+        for sink in self.sink_manager.get_sinks():
+            start = time()
+            sink.send_data(
+                dst=BROADCAST_ADDRESS,
+                src_ep=RTC_SOURCE_EP,
+                dst_ep=RTC_DEST_EP,
+                qos=0,
+                initial_time=INITIAL_DELAY_MS,
+                data=data_payload,
+                is_unack_csma_ca=False,
+                hop_limit=0,
+            )
+            logging.debug("time elapsed to send RTC time through the gateway: "
+                          f"{int((time()-start)*1000)}ms")
+
+    def run(self):
+        """
+        Main loop that send periodically gateway time to the network.
+        """
+        self.running = True
+
+        while self.running:
+            self.publish_time()
+            sleep(self.period)
+
+    def stop(self):
+        """
+        Stop the periodical sending gateway status thread.
+        """
+        self.running = False
+
+class RtcService(BusClient):
+    """
+    Implementation of the RTC service
+
+    Send periodically RTC time to the network.
+    """
+    def __init__(self, settings, **kwargs):
+
+        super(RtcService, self).__init__(**kwargs)
+        self.synchronization_thread = SynchronizationThread(
+            settings.rtc_synchronization_period_s,
+            settings.timezone_offset_s,
+            settings.timezone_from_gateway_clock,
+            settings.get_time_from_local,
+            self.sink_manager,
+        )
+        self.synchronization_thread.start()
+
+    def on_stack_started(self, name):
```

Comment:
Is it really needed to override those calls for just a debug message?

Reply:
Logs are useful but, you are right. If the stack starts, at least the transport service should logs this message. We don't need to duplicate the logs.